File: app/src/main/java/com/utynote/components/map/MapFragment.java
package com.utynote.components.map;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;
import com.utynote.R;

public class MapFragment extends Fragment implements OnMapReadyCallback {
    public static final String TAG = MapFragment.class.getSimpleName();

    private GoogleMap mMap;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle bundle) {
        View view = inflater.inflate(R.layout.map_container_view, container, false);

        FloatingActionButton fab = (FloatingActionButton) view.findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Searching your location..", Snackbar.LENGTH_SHORT)
                        .setAction("Action", null).show();
            }
        });
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        com.google.android.gms.maps.SupportMapFragment mapFragment = new com.google.android.gms.maps.SupportMapFragment();
        mapFragment.getMapAsync(this);
        getChildFragmentManager()
                .beginTransaction()
                .add(R.id.map_container, mapFragment)
                .commit();
    }

    @Override
    public void onMapReady(GoogleMap googleMap) {
        mMap = googleMap;
        LatLng sydney = new LatLng(52.53174, 13.38771);
        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker"));
        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(sydney, 18));
    }
}

File: app/src/main/java/com/utynote/components/map/MapFragment.kt
package com.utynote.components.map

import android.content.Context
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions
import com.utynote.R

class MapFragment : Fragment(), OnMapReadyCallback {

    private var map: GoogleMap? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, bundle: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.map_container_view, container, false)

        (view.findViewById(R.id.fab) as FloatingActionButton)
                .setOnClickListener { v -> Snackbar
                    .make(v, "Searching your location..", Snackbar.LENGTH_SHORT)
                    .setAction("Action", null)
                    .show()
        }
        return view
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        val mapFragment = com.google.android.gms.maps.SupportMapFragment()
        mapFragment.getMapAsync(this)
        childFragmentManager
                .beginTransaction()
                .add(R.id.map_container, mapFragment)
                .commit()
    }

    override fun onMapReady(googleMap: GoogleMap) {
        map = googleMap
        val berlin = LatLng(52.53174, 13.38771)
        googleMap.addMarker(MarkerOptions().position(berlin).title("Marker"))
        googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(berlin, 18f))
    }

    companion object {
        val TAG = MapFragment::class.java.simpleName
    }
}
